### PR TITLE
Move Ruby Version config to anyway_config

### PR DESCRIPTION
**Pull request by colby-swandale**
The current ruby configuration is a painpoint. We have to maintain different configuration options to indicate with versions of Ruby are EOL, which version is the "default", and perform calculations to know which set is currently being maintained by the core team.

This PR introduces the gem [anyway_config](https://github.com/palkan/anyway_config) that moves this configuration into a single YML file that we can compile into a RubyVersion that we easily filter and lookup.

### CODE REVIEW

File: config/ruby.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+default: &default
+  versions:
+    - version: 3.1
+      default: true
```
Comment (AlexWayfer) [inline config/ruby.yml:4]: I'd take `default` out of `versions` to escape multiple `default` (occasionally).

And probably add a check to including `default` in `versions`.

Comment (natematykiewicz) [reply config/ruby.yml:4]: Is there any reason to not just have the first version in the list be the default?

Comment (AlexWayfer) [reply config/ruby.yml:4]: > Is there any reason to not just have the first version in the list be the default?

Not sure. Probably if we'll have some beta or `head` version docs. But until not — it's an option too.

Comment (natematykiewicz) [reply config/ruby.yml:4]: There already is. It's called "dev" and it's the last version in this YAML file.

Comment (AlexWayfer) [reply config/ruby.yml:4]: > There already is. It's called "dev" and it's the last version in this YAML file.

Oh, I'm sorry. In this case, I'm not sure that we should place it as the latest one (excluding obsolete). Like… it's obviously more than latest (3.1 for now), but it's under 2.6 and above 2.5.

**Pull request merged**